lineapy/graph_reader/session_artifacts.py

```diff
@@ -605,9 +607,132 @@ def _update_nodecollection_dependencies(self):
 
     def _get_first_artifact_name(self) -> Optional[str]:
         """
-        Return the name of first artifact(topologically sorted)
+        Return the name of first artifact(topologically sorted).
         """
         for coll in self.artifact_nodecollections:
             if coll.collection_type == NodeCollectionType.ARTIFACT:
                 return coll.safename
         return None
+
+    def get_session_module_imports(self, indentation=0) -> str:
+        """
+        Return all the import statement for the session.
+        """
+        return self.import_nodecollection.get_import_block(
+            indentation=indentation
+        )
+
+    def get_session_function_name(self) -> str:
+        """
+        Return the session function name: run_session_including_{first_artifact_name}
+        """
```

Comment:
To make the code easier to maintain, can we add this type of concrete example to description of other code-generating methods as well? That is, given the abstract nature of these methods, it would be helpful to know what type of output each method is handling/generating. The description may go like,

```python
"""
[SOME HIGH-LEVEL DESCRIPTION].
For instance, if the target session function looks like [EXAMPLE SESSION FUNCTION],
this method is responsible for generating [SUBPART OF EXAMPLE SESSION FUNCTION].
"""
```